app/controllers/submissions_controller.rb

```diff
@@ -124,6 +124,8 @@ def populate_file_manager
       end
       entries = get_all_file_data(revision, grouping, '')
     end
+    entries.append({ only_required_files: assignment.only_required_files,
```

Comment:
Overall, it's not good to add an extra element of a different structure to an otherwise homogeneous list. In this case, a better approach would be to change the `json` response to be an object where one of the keys is the previous `entries`, and the other keys are `only_required_files` and `required_files`. Then the React component can extract each of these three keys in the response object.